app/controllers/pieces_controller.rb

```diff
@@ -4,22 +4,24 @@ def update
     @piece = Piece.find(params[:id])
     @game = Game.find(@piece.game_id)
 
-    if @piece.valid_move?(params[:y_coord].to_i, params[:x_coord].to_i)
+    if @piece.valid_move?(params[:y_coord].to_i, params[:x_coord].to_i) && !@piece.moving_into_check?(params[:y_coord].to_i, params[:x_coord].to_i) 
```

Comment:
After testing this on the board, I get the following bug on my side. Is anyone else getting this?; 
if white player moves to position to where it would put the black king in check, get error message (not good).  
If white player moves to position to where it would put the white king in check, get error message (good).
 If black player, moves to position where white king would be in check, I get a valid move. (good).
If black player moves to position where black king would be in check, I get a valid move (not good).